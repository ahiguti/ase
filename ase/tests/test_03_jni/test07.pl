
my $Comparable = $ASE->{Java}->{"java.lang.Comparable"};
my $Arrays = $ASE->{Java}->{"java.util.Arrays"};

sub Foo::new {
  return bless { value => $_[1] }, $_[0];
}
sub Foo::compareTo {
  my ($x, $y) = @_;
  print "compareTo: $x->{value} $y->{value}\n";
  if ($x->{value} < $y->{value}) { return -1; }
  if ($x->{value} > $y->{value}) { return 1; }
  return 0;
}

my $rndval = 999;
my $len = 20;
my $arr = $Comparable->NewArray($len);
for (my $i = 0; $i < $len; ++$i) {
  $arr->{$i} = $Comparable->NewInstance(new Foo($rndval));
  $rndval = ($rndval * 69069 + 1) % 2147483648;
}
for (my $i = 0; $i < $len; ++$i) {
  my $v = $arr->{$i}->{value};
  print "i=$i v=$v\n";
}

$Arrays->sort($arr);

for (my $i = 0; $i < $len; ++$i) {
  my $v = $arr->{$i}->{value};
  print "i=$i v=$v\n";
}

