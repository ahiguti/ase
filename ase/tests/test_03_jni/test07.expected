i=0 v=999
i=1 v=68999932
i=2 v=490088397
i=3 v=1278232618
i=4 v=1048439715
i=5 v=1534064776
i=6 v=1624304873
i=7 v=272534422
i=8 v=985818399
i=9 v=1374457044
i=10 v=711428549
i=11 v=1085100994
i=12 v=1808723035
i=13 v=1125049312
i=14 v=1482611297
i=15 v=1869401262
i=16 v=221429079
i=17 v=1654000044
i=18 v=441416381
i=19 v=362668634
compareTo: 999 68999932
compareTo: 68999932 490088397
compareTo: 490088397 1278232618
compareTo: 1278232618 1048439715
compareTo: 490088397 1048439715
compareTo: 1534064776 1624304873
compareTo: 1624304873 272534422
compareTo: 1534064776 272534422
compareTo: 1624304873 985818399
compareTo: 1534064776 985818399
compareTo: 272534422 985818399
compareTo: 1624304873 1374457044
compareTo: 1534064776 1374457044
compareTo: 985818399 1374457044
compareTo: 1278232618 272534422
compareTo: 999 272534422
compareTo: 68999932 272534422
compareTo: 490088397 272534422
compareTo: 490088397 985818399
compareTo: 1048439715 985818399
compareTo: 1048439715 1374457044
compareTo: 1278232618 1374457044
compareTo: 711428549 1085100994
compareTo: 1085100994 1808723035
compareTo: 1808723035 1125049312
compareTo: 1085100994 1125049312
compareTo: 1808723035 1482611297
compareTo: 1125049312 1482611297
compareTo: 1869401262 221429079
compareTo: 1869401262 1654000044
compareTo: 221429079 1654000044
compareTo: 1869401262 441416381
compareTo: 1654000044 441416381
compareTo: 221429079 441416381
compareTo: 1869401262 362668634
compareTo: 1654000044 362668634
compareTo: 441416381 362668634
compareTo: 221429079 362668634
compareTo: 1808723035 221429079
compareTo: 711428549 221429079
compareTo: 711428549 362668634
compareTo: 711428549 441416381
compareTo: 711428549 1654000044
compareTo: 1085100994 1654000044
compareTo: 1125049312 1654000044
compareTo: 1482611297 1654000044
compareTo: 1808723035 1654000044
compareTo: 1808723035 1869401262
compareTo: 1624304873 221429079
compareTo: 999 221429079
compareTo: 68999932 221429079
compareTo: 272534422 221429079
compareTo: 272534422 362668634
compareTo: 490088397 362668634
compareTo: 490088397 441416381
compareTo: 490088397 711428549
compareTo: 985818399 711428549
compareTo: 985818399 1085100994
compareTo: 1048439715 1085100994
compareTo: 1278232618 1085100994
compareTo: 1278232618 1125049312
compareTo: 1278232618 1482611297
compareTo: 1374457044 1482611297
compareTo: 1534064776 1482611297
compareTo: 1534064776 1654000044
compareTo: 1624304873 1654000044
i=0 v=999
i=1 v=68999932
i=2 v=221429079
i=3 v=272534422
i=4 v=362668634
i=5 v=441416381
i=6 v=490088397
i=7 v=711428549
i=8 v=985818399
i=9 v=1048439715
i=10 v=1085100994
i=11 v=1125049312
i=12 v=1278232618
i=13 v=1374457044
i=14 v=1482611297
i=15 v=1534064776
i=16 v=1624304873
i=17 v=1654000044
i=18 v=1808723035
i=19 v=1869401262
